src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototypeAzure.cs

```diff
@@ -91,58 +91,58 @@ public string AzureEditionDisplay
+                {
+                    //Try to get the ServiceLevelObjective from the api,if not the default hardcoded service level objectives will be retrieved.
+                    // string serverLevelObjective = AzureServiceLevelObjectiveProvider.TryGetAzureServiceLevelObjective(value, AzureServiceLocation);
+
+                    // if (!string.IsNullOrEmpty(serverLevelObjective))
+                    // {
+                    //     this.currentState.azureEdition = edition;
+                    //     this.currentState.currentServiceLevelObjective = serverLevelObjective;
+                    //     // Instead of creating db instance with default Edition, update EditionToCreate while selecting Edition from the UI.
+                    //     this.EditionToCreate = MapAzureEditionToDbEngineEdition(edition);
+                    //     string storageAccountType = AzureServiceLevelObjectiveProvider.TryGetAzureStorageType(value, AzureServiceLocation);
+                    //     if (!string.IsNullOrEmpty(storageAccountType))
+                    //     {
+                    //         this.currentState.backupStorageRedundancy = storageAccountType;
+                    //     }
+
+                    //     // Try to get the azure maxsize from the api,if not the default hardcoded maxsize will be retrieved.
+                    //     DbSize dbSize = AzureServiceLevelObjectiveProvider.TryGetAzureMaxSize(value, serverLevelObjective, AzureServiceLocation);
+                    //     if (!string.IsNullOrEmpty(dbSize.ToString()))
+                    //     {
+                    //         this.currentState.maxSize = new DbSize(dbSize.Size, dbSize.SizeUnit);
+                    //     }
+                    // }
+                    // else
+                    // {
+                    if (edition == this.currentState.azureEdition)
+                    { // No changes, return early since we don't need to do any of the changes below
+                        return;
+                    }
+
+                    this.currentState.azureEdition = edition;                        
+                    this.EditionToCreate = MapAzureEditionToDbEngineEdition(edition);
+                    this.CurrentServiceLevelObjective = AzureSqlDbHelper.GetDefaultServiceObjective(edition);
+                    this.BackupStorageRedundancy = AzureSqlDbHelper.GetDefaultBackupStorageRedundancy(edition);
+                    var defaultSize = AzureSqlDbHelper.GetDatabaseDefaultSize(edition);
+
+                    this.MaxSize = defaultSize == null ? String.Empty : defaultSize.ToString();
+                    // }
+                    this.NotifyObservers();
+                }
+                else
+                {
+                    //Can't really do much if we fail to parse the display name so just leave it as is and log a message
+                    System.Diagnostics.Debug.Assert(false,
```

Comment:
I'm going to keep bugging you about this every time I see these - there is 0 good reason to keep a debug assert in this code vs having an exception or just default to some known good value. This one just seems like it should be logged and move on.